src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -55,22 +77,50 @@ protected void doRun() {
 
     private void cleanCloud(ComputeEngineCloud cloud) {
         logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
-        List<Instance> remoteInstances = findRemoteInstances(cloud);
+        ComputeClientV2 clientV2;
+        try {
+            clientV2 = cloud.getClientV2();
+        } catch (GeneralSecurityException | IOException ex) {
+            logger.log(Level.WARNING, "Error getting clientV2 for cloud " + cloud.getCloudName(), ex);
+            return;
+        }
+        List<Instance> remoteInstances = findRunningRemoteInstances(clientV2);
         Set<String> localInstances = findLocalInstances(cloud);
+        if (!(localInstances.isEmpty() || remoteInstances.isEmpty())) {
+            updateLocalInstancesLabel(clientV2, localInstances, remoteInstances);
+        }
         remoteInstances.stream()
                 .filter(remote -> isOrphaned(remote, localInstances))
                 .forEach(remote -> terminateInstance(remote, cloud));
     }
 
     private boolean isOrphaned(Instance remote, Set<String> localInstances) {
-        String instanceName = remote.getName();
-        logger.log(Level.FINEST, "Checking instance " + instanceName);
-        return !localInstances.contains(instanceName);
+        /* It is necessary to check if the remote instance is present in localInstances.
+           The `remote` instance has an old timestamp because it hasn't been fetched again
+           after the `updateLocalInstancesLabel` call, to avoid extra network calls.
+        */
+        if (localInstances.contains(remote.getName())) {
+            return false;
+        }
+        String nodeLastRefresh = remote.getLabels().get(NODE_IN_USE_LABEL_KEY);
+        if (nodeLastRefresh == null) {
+            return false;
+        }
+        OffsetDateTime lastRefresh =
+                LocalDateTime.parse(nodeLastRefresh, formatter).atOffset(ZoneOffset.UTC);
+        boolean isOrphan = lastRefresh
+                .plus(RECURRENCE_PERIOD * LOST_MULTIPLIER, ChronoUnit.MILLIS)
+                .isBefore(OffsetDateTime.now(ZoneOffset.UTC));
+        logger.log(
+                Level.FINEST,
+                "Instance " + remote.getName() + " last_refresh label value: " + nodeLastRefresh + ", isOrphan: "
+                        + isOrphan);
```

Comment:
You shouldn't do inline string concatenation for a log level that will rarely be active
```suggestion
        logger.log(
                Level.FINEST, () ->
                "Instance " + remote.getName() + " last_refresh label value: " + nodeLastRefresh + ", isOrphan: "
                        + isOrphan);
```
or
```suggestion
        if (logger.isLoggable(Level.FINEST)) {
            logger.log(Level.FINEST,
                "Instance " + remote.getName() + " last_refresh label value: " + nodeLastRefresh + ", isOrphan: "
                        + isOrphan);
       }                        
```

Reply:
noted about it. thank you so much., already applied. will follow in future as well